test/src/unit-cppapi-schema-evolution.cc

```diff
@@ -107,6 +107,50 @@ TEST_CASE(
   }
 }
 
+TEST_CASE(
+    "C++ API: SchemaEvolution, check error when dropping dimension",
+    "[cppapi][schema][evolution][drop]") {
+  using namespace tiledb;
+  Context ctx;
+  VFS vfs(ctx);
+
+  std::string array_uri = "test_schema_evolution_array";
+
+  Domain domain(ctx);
+  auto id1 = Dimension::create<int>(ctx, "d1", {{-100, 100}}, 10);
+  auto id2 = Dimension::create<int>(ctx, "d2", {{0, 100}}, 5);
+  domain.add_dimension(id1).add_dimension(id2);
+
+  auto a1 = Attribute::create<int>(ctx, "a1");
+  auto a2 = Attribute::create<int>(ctx, "a2");
+
+  ArraySchema schema(ctx, TILEDB_DENSE);
+  schema.set_domain(domain);
+  schema.add_attribute(a1);
+  schema.add_attribute(a2);
+  schema.set_cell_order(TILEDB_ROW_MAJOR);
+  schema.set_tile_order(TILEDB_COL_MAJOR);
+
+  if (vfs.is_dir(array_uri)) {
+    vfs.remove_dir(array_uri);
+  }
+
+  Array::create(array_uri, schema);
+
+  auto evolution = ArraySchemaEvolution(ctx);
+
+  // try to drop dimension d1
+  evolution.drop_attribute("d1");
```

Comment:
This is the statement that should fail. Avoiding a failure here only defers detection of the problem until later.

Reply:
> the ArraySchema object [...] is not available in drop_attribute()

Yes, I see that now. I didn't realize the whole design was broken in this particular way.